app/src/main/kotlin/com/google/samples/apps/nowinandroid/MainActivity.kt

```diff
@@ -81,58 +77,64 @@ class MainActivity : ComponentActivity() {
     lateinit var userNewsResourceRepository: UserNewsResourceRepository
 
     val viewModel: MainActivityViewModel by viewModels()
-
     override fun onCreate(savedInstanceState: Bundle?) {
         val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
 
-        var uiState: MainActivityUiState by mutableStateOf(Loading)
+        // We keep this as a mutable state, so that we can track changes inside the composition.
+        // This allows us to react to dark/light mode changes.
+        var themeSettings by mutableStateOf(
+            ThemeSettings(
+                darkTheme = resources.configuration.isSystemInDarkTheme,
+                androidTheme = Loading.shouldUseAndroidTheme,
+                disableDynamicTheming = Loading.shouldDisableDynamicTheming,
+            ),
+        )
 
         // Update the uiState
         lifecycleScope.launch {
             lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                viewModel.uiState
-                    .onEach { uiState = it }
-                    .collect()
+                combine(
+                    isSystemInDarkTheme(),
+                    viewModel.uiState,
+                ) { systemDark, uiState ->
+                    ThemeSettings(
+                        darkTheme = uiState.shouldUseDarkTheme(systemDark),
+                        androidTheme = uiState.shouldUseAndroidTheme,
+                        disableDynamicTheming = uiState.shouldDisableDynamicTheming,
+                    )
+                }
+                    .onEach { themeSettings = it }
+                    .map { it.darkTheme }
+                    .distinctUntilChanged()
```

Comment:
My thinking was that I don't want the `enableEdgeToEdge` to be called when unrelated things change (`androidTheme` or `disableDynamicTheming`). Althought `isSystemInDarkTheme` and `uiState` deduplicate emissions, the `combine` of them don't. We don't care when setting the `themeSettings`, because `mutableStateOf` automatically uses `structuralEqualityPolicy`, but then the param doesn't. 

Maybe it's over-complicated though 😅

Reply:
My question comes from the fact that each `Flow` only emits the latest values after doing deduplication upstream. Considering this, is there still a point in doing deduplication downstream in the `combine` function? 
Since each Flow already ensures only the latest values are emitted, deduplicating again downstream seems redundant to me.

```kotlin
val ms = MutableStateFlow(1)

launch {
    delay(5000)
    ms.update { it + 1 }
}

val fs = flow {
    emit(true)
    delay(2000)
    emit(true)
    delay(2000)
    emit(false)
    delay(2000)
    emit(false)
    delay(2000)
    emit(false)
}
    .onStart { println("Start") }
    .onCompletion { println("End") }
    .distinctUntilChanged()

combine(fs, ms) { f, m -> "$f, $m" }
    .onEach { println(it) }
    .collect()

// Start
// true, 1
// false, 1
// false, 2
// End
```

Could I be misunderstanding something here? 🥲

Reply:
Hm, I think your example is not complete.

Let's consider this:
```
combine(fs, ms) { f, m -> f to m }
    .onEach { println("set theme $it") }
    .onEach { println("enable edge to edge ${it.second}") }
    .collect()
```

This will produce
```
Start
set theme (true, 1)
enable edge to edge 1
set theme (false, 1)
enable edge to edge 1 <---- unnecessary call
set theme (false, 2)
enable edge to edge 2
End
```

While if you do the `map` + `distinct`

```
combine(fs, ms) { f, m -> f to m }
    .onEach { println("set theme $it") }
    .map { it.second }
    .distinctUntilChanged()
    .onEach { println("enable edge to edge $it") }
    .collect()
```

```
Start
set theme (true, 1)
enable edge to edge 1
set theme (false, 1)
set theme (false, 2)
enable edge to edge 2 <-- called only when "m" changed
End
```

So the reason is to skip calling `enableEdgeToEdge` if it didn't change.
The reason is that this call doesn't have internal check if it has the same parameters, but the resulting change is short, so maybe it's just overcomplicated.

Reply:
Ah.. I completely understand, thank you

Reply:
Thanks for the question!